embedded/remotestorage/s3/s3.go

```diff
@@ -73,12 +80,20 @@ var (
 	ErrInvalidResponseSubPathUnescape      = fmt.Errorf("%w: error un-escaping object name", ErrInvalidResponse)
 
 	ErrTooManyRedirects = errors.New("too many redirects")
+
+	arnRoleRegex = regexp.MustCompile(`arn:.*\/(.*)`)
 )
 
-const maxRedirects = 5
+const (
+	awsCredsRefreshPeriod  = time.Minute
+	awsInstanceMetadataURL = "http://169.254.169.254"
```

Comment:
this type of constant definitions should not be part of the code  `awsInstanceMetadataURL = "http://169.254.169.254"`

Reply:
actually, the 169.254.169.254 is a fixed ip address used by aws instances to get their runtime parameters and/or as api endpoint. So I think it makes sense to hardwire it in the code

Reply:
@SimoneLazzaris you are right. As for now, I've made this into a default value for a parameter (flag or env var), so should be used anyway, just with a bit more flexibility than initially required.

Would you please push this a bit further to review or possibly merge this one?